[ADDON] Immich Public Proxy: Service fails to start (Invalid ExecStart path)

### ✅ Have you read and understood the above guidelines?

yes

### 🔎 Did you run the script with verbose mode enabled?

Yes, verbose mode was enabled and the output is included below

### 📜 What is the name of the script you are using?

 Immich Public Proxy

### 📂 What was the exact command used to execute the script?

bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/tools/addon/immich-public-proxy.sh)"

### ⚙️ What settings are you using?

- [x] Default Settings
- [ ] Advanced Settings

### 🖥️ Which Linux distribution are you using?

Debian 13

### 📈 Which Proxmox version are you on?

9.0.10

### 📝 Provide a clear and concise description of the issue.

I installed the Immich Public Proxy addon via the community script. The installation completes, but the systemd service fails to start immediately.

Checking the logs reveals that the service is trying to execute /opt/immich-proxy/app/server.js, which does not exist.

The upstream repository (alangrainger/immich-public-proxy) is a TypeScript project. The entry point after building is located at dist/index.js, not server.js in the root.

### 🔄 Steps to reproduce the issue.

1. Run the Immich Public Proxy addon script in an Immich LXC.
2. Observe that the service immich-proxy fails to start.
3. Check systemctl status immich-proxy or journalctl -u immich-proxy.

### ❌ Paste the full error output (if available).

```
Feb 17 09:20:43 immich systemd[1]: Started immich-proxy.service - Immich Public Proxy.
Feb 17 09:20:43 immich node[1056616]: Error: Cannot find module '/opt/immich-proxy/app/server.js'
Feb 17 09:20:43 immich node[1056616]:     at Module._resolveFilename (node:internal/modules/cjs/loader:1420:15)
Feb 17 09:20:43 immich node[1056616]:     at Module._load (node:internal/modules/cjs/loader:1226:37)
Feb 17 09:20:43 immich node[1056616]:   code: 'MODULE_NOT_FOUND',
Feb 17 09:20:43 immich systemd[1]: immich-proxy.service: Main process exited, code=exited, status=1/FAILURE
```

### 🖼️ Additional context (optional).

Current (Broken) Service Config:
```
ExecStart=/usr/bin/node /opt/immich-proxy/app/server.js
```
Possible Fix:
```
ExecStart=/usr/bin/node /opt/immich-proxy/app/dist/index.js
```

--- a/tools/addon/immich-public-proxy.sh
+++ b/tools/addon/immich-public-proxy.sh
@@ -104,12 +104,37 @@ function update() {
     $STD npm run build
     msg_ok "Built ${APP}"
 
+    msg_info "Updating service"
+    create_service
+    msg_ok "Updated service"
+
     msg_info "Starting service"
     systemctl start immich-proxy
     msg_ok "Started service"
     msg_ok "Updated successfully"
     exit
   fi
+}
+
+function create_service() {
+  cat <<EOF >"$SERVICE_PATH"
+[Unit]
+Description=Immich Public Proxy
+After=network.target
+
+[Service]
+Type=simple
+User=root
+WorkingDirectory=${INSTALL_PATH}/app
+EnvironmentFile=${CONFIG_PATH}/.env
+ExecStart=/usr/bin/node ${INSTALL_PATH}/app/dist/index.js
+Restart=always
+RestartSec=10
+
+[Install]
+WantedBy=multi-user.target
+EOF
+  systemctl daemon-reload
 }
 
 # ==============================================================================
@@ -173,23 +198,7 @@ EOF
   msg_ok "Created configuration"
 
   msg_info "Creating service"
-  cat <<EOF >"$SERVICE_PATH"
-[Unit]
-Description=Immich Public Proxy
-After=network.target
-
-[Service]
-Type=simple
-User=root
-WorkingDirectory=${INSTALL_PATH}
-EnvironmentFile=${CONFIG_PATH}/.env
-ExecStart=/usr/bin/node ${INSTALL_PATH}/app/server.js
-Restart=always
-RestartSec=10
-
-[Install]
-WantedBy=multi-user.target
-EOF
+  create_service
   systemctl enable -q --now immich-proxy
   msg_ok "Created and started service"
 

